### PR TITLE
move database configuration to system env variables

### PR DESCRIPTION
**Pull request by azhi**
this will allow us to be hosted on heroku

Ecto.Repo supports database URLs from env, so we don't need to do
any extra work

use separate env variable for test database (we could keep default
config, but better use env vars everywhere for consistency)

### CODE REVIEW

**Comment by azhi** [comment]
[Think of and document a way to manage backend configuration (database credentials).](https://trello.com/c/vZwvaqOa/3-think-of-and-document-a-way-to-manage-backend-configuration-database-credentials)

**Comment by gekola** [comment]
I hoped github can do rebases automatically, too bad it can't. :disappointed: 

```
** (KeyError) key :database not found in: [otp_app: :poscaster, repo: Poscaster.Repo, adapter: Ecto.Adapters.Postgres, pool: Ecto.Adapters.SQL.Sandbox]
    (elixir) lib/keyword.ex:333: Keyword.fetch!/2
    lib/ecto/adapters/postgres.ex:88: Ecto.Adapters.Postgres.storage_up/1
    ...
```
[^ this](https://travis-ci.org/poscaster/backend/builds/168727990#L336) has to be addressed before merge.

**Comment by gekola** [comment]
Also, if we proceed with `source .env && mix test` (or `env $(cat .env | xargs) mix test`) approach README **have to** be updated accordingly.

**Comment by azhi** [comment]
Updated, moved dev & test database config back to plain elixir. Added example secret files that are imported in configs.

File: config/dev.exs [hunk 1]
```diff
@@ -13,8 +13,7 @@ config :phoenix, :stacktrace_depth, 20
 
 config :poscaster, Poscaster.Repo,
   adapter: Ecto.Adapters.Postgres,
-  username: "postgres",
-  password: "postgres",
-  database: "poscaster_dev",
-  hostname: "localhost",
+  url: {:system, "DATABASE_URL"},
```
Comment (gekola) [inline config/dev.exs]: Won't these two conflict?

Comment (azhi) [inline config/dev.exs]: They won't (last one will be used), but that's unnecessary leftover anyway. I've already removed it.

**Pull request merged**